packages/replay/src/util/addEvent.ts

```diff
@@ -37,12 +39,27 @@ export async function addEvent(
   // Only record earliest event if a new session was created, otherwise it
   // shouldn't be relevant
   const earliestEvent = replay.getContext().earliestEvent;
-  if (replay.session && replay.session.segmentId === 0 && (!earliestEvent || timestampInMs < earliestEvent)) {
+  if (session.segmentId === 0 && (!earliestEvent || timestampInMs < earliestEvent)) {
     replay.getContext().earliestEvent = timestampInMs;
   }
 
   try {
-    return await replay.eventBuffer.addEvent(event, isCheckout);
+    if (isCheckout) {
+      if (replay.recordingMode === 'error') {
+        // Do not wait on it, just do it
+        // We know in this mode this is actually "sync"
+        void eventBuffer.clear(true);
+
+        // Ensure we have the correct first checkout timestamp when an error occurs
+        if (!session.segmentId) {
+          replay.getContext().earliestEvent = eventBuffer.getEarliestTimestamp();
+        }
```

Comment:
Just to clarify for myself, we examine the buffer for the earliest event because we want the replay "start time" to be when the 1st checkout in the buffer happens.

So if session has been running for 2.5 minutes, we would want the replay to "start" at ~2 minutes. And `replay.context.earliestEvent` (before this block of code), would represent the earliest event of the entire "session".


So an issue I can think of is with the performance observer. I believe we only add them when we flush, so those events will end up becoming the earliest events anyway (which might be fine).

Reply:
Yeah, to be honest all of this is a bit brittle to me 😅 but this was the best I could come up with to handle this in a somewhat reasonable way. I wonder if it makes sense to maybe compute this on the server? Or generally change how we keep this a bit, not sure...

Reply:
semi-related to this PR but I was also wondering about that - do we discard events with timestamps before the initial full checkout on the server? Or would sending such events cause problems?

Reply:
semi-related to this PR but I was also wondering about that - do we discard events with timestamps before the initial full checkout on the server? Or would sending such events cause problems?

Reply:
We discard events that are older than SESSION_IDLE_TIMEOUT minutes ago. Because we get buffered data from performance observer, those occur before initial checkout.